analyzer/src/types/ctx.rs

```diff
@@ -0,0 +1,69 @@
+use crate::relations::{Relations, SourceObjectId, Symbol};
+use crate::types::hir::TypeId;
+use crate::types::{BOOL, FLOAT, INT, NOTHING, STRING};
+use std::collections::HashMap;
+
+pub struct TypeContext {
+    pub(crate) source: SourceObjectId,
```

Comment:
```suggestion
    pub(crate) current_source: SourceObjectId,
```
first thought it was a context for each sources :p